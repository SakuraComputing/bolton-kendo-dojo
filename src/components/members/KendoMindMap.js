import React, { Component } from 'react';

class KendoMindMap extends Component {

  render() {
    return (
      <div className="content-container mindmap-layout">
        <img src="/img/background/KendoMindMap.png" alt=""/>
      </div>
    )
  }
}
export default KendoMindMap;
