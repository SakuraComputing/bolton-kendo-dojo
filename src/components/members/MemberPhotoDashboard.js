import React from 'react'
import MemberPhotoFilter from './MembersPhotoFilter';
import MemberPhotos from './MembersPhotos';

const MemberPhotoDashboardPage = () => (
    <div>
        <MemberPhotoFilter />
        <MemberPhotos />        
    </div>

)
export default MemberPhotoDashboardPage;