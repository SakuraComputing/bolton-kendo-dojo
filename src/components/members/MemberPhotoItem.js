import React from 'react';

const MemberPhotoItem = (props) => {
    return (
        <div className="image-frame">
            <img className="image-item" src={`uploads/${props.filename}`} alt="photoU" />
            <div>
                {props.description}
            </div>
            <button onClick={props.onUploadDelete} className="button-small">Delete Photo</button>
        </div>
    );
}

export default MemberPhotoItem;
