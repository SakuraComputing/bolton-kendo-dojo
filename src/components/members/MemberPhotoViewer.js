import React from 'react'

const MemberPhotoViewer = () => {
    return (
        <div>
            <h1>Photo Viewer</h1>
        </div>
    )
}

export default MemberPhotoViewer
