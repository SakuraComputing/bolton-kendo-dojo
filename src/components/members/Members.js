import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photos</Link>
          <p>Select the photographs and upload these with text for other members to see.</p>
        </div>
        <div className="members-area-link">
          <Link to="/mindmap">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Respond to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
