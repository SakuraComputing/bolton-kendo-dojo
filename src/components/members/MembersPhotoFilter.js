import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setPhotoTextFilter, setPhotoFilterTitle, setPhotoDateSortOrder } from '../../actions/filterActions';

export class MembersPhotoFilter extends Component {

    onTextChange = (e) => {
        this.props.setPhotoTextFilter(e.target.value);
    }

    onTitleChange = (e) => {
        this.props.setPhotoFilterTitle(e.target.value);
    }

    render() {
        return (
            <div>
                <div className="wrap-collapsible">
                        <input id="collapsible-search" className="toggle-search" type="checkbox"/>
                        <label htmlFor="collapsible-search" className="lbl-toggle-search">Search</label>
                        <div className="collapsible-content-search">
                            <div className="content-inner">
                                <form onSubmit={this.onFormSubmit} className="image-header">
                                    <div className="image-search-container">
                                        <input
                                            type="text"
                                            className="filter-text input-box"
                                            placeholder="Search By Description"
                                            value={this.props.filters.photoText}
                                            onChange={this.onTextChange}
                                            name="filterText"
                                        />                                        
                                        <input
                                            type="text"
                                            className="filter-title input-box"
                                            placeholder="Search By Title"
                                            value={this.props.filters.photoTitle}
                                            onChange={this.onTitleChange}
                                            name="filterTitle"
                                        />                                        
                                    </div>
                                </form>
                            </div>
                        </div>
                    </div>

            </div>
        )
    }
}

const mapStateToProps = state => ({
    filters: state.filters
})

const mapDispatchToProps = (dispatch) => ({
    setPhotoTextFilter: (photoText) => dispatch(setPhotoTextFilter(photoText)),
    setPhotoFilterTitle: (photoTitle) => dispatch(setPhotoFilterTitle(photoTitle)),
    setPhotDateSortOrder: () => dispatch(setPhotoDateSortOrder())
})
export default connect(mapStateToProps, mapDispatchToProps)(MembersPhotoFilter);
