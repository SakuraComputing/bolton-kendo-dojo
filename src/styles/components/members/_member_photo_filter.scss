.image-search-container {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.filter-text {
    outline: none;
}

.filter-title {
    outline: none;
    margin-left: 10px;
}