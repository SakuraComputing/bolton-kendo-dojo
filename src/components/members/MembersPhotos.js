import React from 'react';
import { connect } from 'react-redux';
import Spinner from '../../common/Spinner';
import { getMemberUploads, postMemberUploads, deleteMemberUploads } from '../../actions/uploadActions';
import MemberPhotoItem from '../../components/members/MemberPhotoItem';

export class MemberPhoto extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            file: null,
            description: '',
            errors: {}
        };
    };

    componentDidMount() {
        this.props.getMemberUploads();
    }

    onChange = (e) => {
        this.setState({ file: e.target.files[0] })  
    };

    onDescriptionChange = (e) => {
        this.setState({ description: e.target.value })
    }

    onFormSubmit = (e) => {
        e.preventDefault();

        const { file, description } = this.state; 

        let formData = new FormData();
        formData.append('image', file);
        formData.set('description', description);
        this.props.postMemberUploads(formData);
    }

    onUploadDelete = (index, e) => {
        const photoId = this.props.uploads.uploads[index]._id;
        this.props.deleteMemberUploads(photoId);
    }

    render() {

        let album;

        const { uploads, loading } = this.props.uploads;

        if(uploads === null || uploads === undefined || loading) {
            album = <Spinner />
        } else {
            if(uploads.length > 0) {
                album = uploads.map(((image, key) => {
                    return <MemberPhotoItem 
                            key={image._id}
                            filename={image.filename}
                            description={image.description}
                            onUploadDelete={this.onUploadDelete.bind(this, key)}
                        />
                }))
            } else {
                album = <h4>No uploads found....</h4>
            }
        }
        
        return (
            <div className="images-uploads content-container">
                <h1 className="image-title">Member Photograph Upload</h1>

                <div className="image-header">
                    <form onSubmit={this.onFormSubmit}>
                        <input type="file" className="upload-file" accept=".jpg,.jpeg,.png" name="image" id="file" onChange={this.onChange} />
                        <label className="upload-label" htmlFor="file">Upload Photo</label>
                        <input type="submit" value="Submit" className="button-small"/>
                        <label className="upload-label" htmlFor="description" >Add Photo Description</label>
                        <input className="upload-text input-box" type="text" name="description" onChange={this.onDescriptionChange}/>
                    </form>
                </div>
                
                <div className="image-container">
                    {album}
                </div>
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    errors: state.errors,
    uploads: state.uploads
})
export default connect(mapStateToProps, { getMemberUploads , postMemberUploads, deleteMemberUploads })(MemberPhoto);
