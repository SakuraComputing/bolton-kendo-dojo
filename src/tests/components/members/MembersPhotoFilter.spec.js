import React from 'react'
import { shallow } from 'enzyme';
import { MembersPhotoFilter } from '../../../components/members/MembersPhotoFilter';
// import filters from '../../fixtures/filters';

describe('Members photo filter component', () => {

    let memberFilter, setPhotoTextFilter, filters;

    beforeEach(() => {
        setPhotoTextFilter = jest.fn();
        filters = jest.fn();
        memberFilter = shallow(<MembersPhotoFilter setPhotoTextFilter={setPhotoTextFilter} filters={filters}/>)
    })
    it('should render correctly', () => {
        expect(memberFilter).toMatchSnapshot();
    });
    describe('when typing into the photo text filter field', () => {
        
        const description = 'More Details'

        beforeEach(() => {
            memberFilter.find('.filter-text').simulate('change', { target: { value: description }});
        })

        it('should set props to equal text photo input', () => {
            expect(setPhotoTextFilter).toHaveBeenCalledWith(description);
        });
    });
    describe('when typing into the photo title filter field', () => {
        
        const title = 'Title1'

        beforeEach(() => {
            memberFilter.find('.filter-title').simulate('change', { target: { value: title }});
        })

        it('should set props to equal text photo input', () => {
            expect(setPhotoTextFilter).toHaveBeenCalledWith(title);
        });
    });
});