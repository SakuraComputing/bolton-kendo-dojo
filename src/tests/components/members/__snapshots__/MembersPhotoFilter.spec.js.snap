// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members photo filter component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MembersPhotoFilter
    filters={[MockFunction]}
    setPhotoFilterTitle={[MockFunction]}
    setPhotoTextFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wrap-collapsible"
      >
        <input
          className="toggle-search"
          id="collapsible-search"
          type="checkbox"
        />
        <label
          className="lbl-toggle-search"
          htmlFor="collapsible-search"
        >
          Search
        </label>
        <div
          className="collapsible-content-search"
        >
          <div
            className="content-inner"
          >
            <form
              className="image-header"
            >
              <div
                className="image-search-container"
              >
                <input
                  className="filter-text input-box"
                  name="filterText"
                  onChange={[Function]}
                  placeholder="Search By Description"
                  type="text"
                />
                <input
                  className="filter-title input-box"
                  name="filterTitle"
                  onChange={[Function]}
                  placeholder="Search By Title"
                  type="text"
                />
              </div>
            </form>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="toggle-search"
            id="collapsible-search"
            type="checkbox"
          />,
          <label
            className="lbl-toggle-search"
            htmlFor="collapsible-search"
          >
            Search
          </label>,
          <div
            className="collapsible-content-search"
          >
            <div
              className="content-inner"
            >
              <form
                className="image-header"
              >
                <div
                  className="image-search-container"
                >
                  <input
                    className="filter-text input-box"
                    name="filterText"
                    onChange={[Function]}
                    placeholder="Search By Description"
                    type="text"
                  />
                  <input
                    className="filter-title input-box"
                    name="filterTitle"
                    onChange={[Function]}
                    placeholder="Search By Title"
                    type="text"
                  />
                </div>
              </form>
            </div>
          </div>,
        ],
        "className": "wrap-collapsible",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "toggle-search",
            "id": "collapsible-search",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "lbl-toggle-search",
            "htmlFor": "collapsible-search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="content-inner"
            >
              <form
                className="image-header"
              >
                <div
                  className="image-search-container"
                >
                  <input
                    className="filter-text input-box"
                    name="filterText"
                    onChange={[Function]}
                    placeholder="Search By Description"
                    type="text"
                  />
                  <input
                    className="filter-title input-box"
                    name="filterTitle"
                    onChange={[Function]}
                    placeholder="Search By Title"
                    type="text"
                  />
                </div>
              </form>
            </div>,
            "className": "collapsible-content-search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                className="image-header"
              >
                <div
                  className="image-search-container"
                >
                  <input
                    className="filter-text input-box"
                    name="filterText"
                    onChange={[Function]}
                    placeholder="Search By Description"
                    type="text"
                  />
                  <input
                    className="filter-title input-box"
                    name="filterTitle"
                    onChange={[Function]}
                    placeholder="Search By Title"
                    type="text"
                  />
                </div>
              </form>,
              "className": "content-inner",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="image-search-container"
                >
                  <input
                    className="filter-text input-box"
                    name="filterText"
                    onChange={[Function]}
                    placeholder="Search By Description"
                    type="text"
                  />
                  <input
                    className="filter-title input-box"
                    name="filterTitle"
                    onChange={[Function]}
                    placeholder="Search By Title"
                    type="text"
                  />
                </div>,
                "className": "image-header",
                "onSubmit": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      className="filter-text input-box"
                      name="filterText"
                      onChange={[Function]}
                      placeholder="Search By Description"
                      type="text"
                    />,
                    <input
                      className="filter-title input-box"
                      name="filterTitle"
                      onChange={[Function]}
                      placeholder="Search By Title"
                      type="text"
                    />,
                  ],
                  "className": "image-search-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "filter-text input-box",
                      "name": "filterText",
                      "onChange": [Function],
                      "placeholder": "Search By Description",
                      "type": "text",
                      "value": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "filter-title input-box",
                      "name": "filterTitle",
                      "onChange": [Function],
                      "placeholder": "Search By Title",
                      "type": "text",
                      "value": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              "type": "form",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrap-collapsible"
        >
          <input
            className="toggle-search"
            id="collapsible-search"
            type="checkbox"
          />
          <label
            className="lbl-toggle-search"
            htmlFor="collapsible-search"
          >
            Search
          </label>
          <div
            className="collapsible-content-search"
          >
            <div
              className="content-inner"
            >
              <form
                className="image-header"
              >
                <div
                  className="image-search-container"
                >
                  <input
                    className="filter-text input-box"
                    name="filterText"
                    onChange={[Function]}
                    placeholder="Search By Description"
                    type="text"
                  />
                  <input
                    className="filter-title input-box"
                    name="filterTitle"
                    onChange={[Function]}
                    placeholder="Search By Title"
                    type="text"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="toggle-search"
              id="collapsible-search"
              type="checkbox"
            />,
            <label
              className="lbl-toggle-search"
              htmlFor="collapsible-search"
            >
              Search
            </label>,
            <div
              className="collapsible-content-search"
            >
              <div
                className="content-inner"
              >
                <form
                  className="image-header"
                >
                  <div
                    className="image-search-container"
                  >
                    <input
                      className="filter-text input-box"
                      name="filterText"
                      onChange={[Function]}
                      placeholder="Search By Description"
                      type="text"
                    />
                    <input
                      className="filter-title input-box"
                      name="filterTitle"
                      onChange={[Function]}
                      placeholder="Search By Title"
                      type="text"
                    />
                  </div>
                </form>
              </div>
            </div>,
          ],
          "className": "wrap-collapsible",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "toggle-search",
              "id": "collapsible-search",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "lbl-toggle-search",
              "htmlFor": "collapsible-search",
            },
            "ref": null,
            "rendered": "Search",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="content-inner"
              >
                <form
                  className="image-header"
                >
                  <div
                    className="image-search-container"
                  >
                    <input
                      className="filter-text input-box"
                      name="filterText"
                      onChange={[Function]}
                      placeholder="Search By Description"
                      type="text"
                    />
                    <input
                      className="filter-title input-box"
                      name="filterTitle"
                      onChange={[Function]}
                      placeholder="Search By Title"
                      type="text"
                    />
                  </div>
                </form>
              </div>,
              "className": "collapsible-content-search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  className="image-header"
                >
                  <div
                    className="image-search-container"
                  >
                    <input
                      className="filter-text input-box"
                      name="filterText"
                      onChange={[Function]}
                      placeholder="Search By Description"
                      type="text"
                    />
                    <input
                      className="filter-title input-box"
                      name="filterTitle"
                      onChange={[Function]}
                      placeholder="Search By Title"
                      type="text"
                    />
                  </div>
                </form>,
                "className": "content-inner",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="image-search-container"
                  >
                    <input
                      className="filter-text input-box"
                      name="filterText"
                      onChange={[Function]}
                      placeholder="Search By Description"
                      type="text"
                    />
                    <input
                      className="filter-title input-box"
                      name="filterTitle"
                      onChange={[Function]}
                      placeholder="Search By Title"
                      type="text"
                    />
                  </div>,
                  "className": "image-header",
                  "onSubmit": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="filter-text input-box"
                        name="filterText"
                        onChange={[Function]}
                        placeholder="Search By Description"
                        type="text"
                      />,
                      <input
                        className="filter-title input-box"
                        name="filterTitle"
                        onChange={[Function]}
                        placeholder="Search By Title"
                        type="text"
                      />,
                    ],
                    "className": "image-search-container",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "filter-text input-box",
                        "name": "filterText",
                        "onChange": [Function],
                        "placeholder": "Search By Description",
                        "type": "text",
                        "value": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "filter-title input-box",
                        "name": "filterTitle",
                        "onChange": [Function],
                        "placeholder": "Search By Title",
                        "type": "text",
                        "value": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                "type": "form",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
